devicetypes/smartthings/zwave-radiator-thermostat.src/zwave-radiator-thermostat.groovy

```diff
@@ -174,9 +177,28 @@ def zwaveEvent(physicalgraph.zwave.commands.thermostatmodev2.ThermostatModeRepor
 	createEvent(map)
 }
 
-def zwaveEvent(physicalgraph.zwave.commands.thermostatsetpointv2.ThermostatSetpointReport cmd) {
+def updateSetpoint(cmd) {
 	def deviceTemperatureScale = cmd.scale ? 'F' : 'C'
-	createEvent(name: "heatingSetpoint", value: convertTemperatureIfNeeded(cmd.scaledValue, deviceTemperatureScale, cmd.precision), unit: temperatureScale)
+	def setpoint = Float.parseFloat(convertTemperatureIfNeeded(cmd.scaledValue, deviceTemperatureScale, cmd.precision))
+	state.cachedSetpoint = setpoint
+	createEvent(name: "heatingSetpoint", value: setpoint, unit: temperatureScale)
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.thermostatsetpointv2.ThermostatSetpointReport cmd) {
+	if (!state.isSetpointChangeRequestedByController) {
+		updateSetpoint(cmd)
+	} else {
+		[:]
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.thermostatsetpointv2.ThermostatSetpointReport cmd, isResponseOfWakeUp) {
+	if (state.isSetpointChangeRequestedByController) {
+		state.isSetpointChangeRequestedByController = false
+		updateSetpoint(cmd)
+	} else {
+		[:]
+	}
```

Comment:
Could you combine these two into one handler with a default value for `isResponseOfWakeUp`?